modules/nf-core/vg/construct/main.nf

```diff
@@ -0,0 +1,68 @@
+process VG_CONSTRUCT {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::vg=1.45.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/vg:1.45.0--h9ee0642_0':
+        'quay.io/biocontainers/vg:1.45.0--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(vcfs), path(tbis), path(msa), path(insertions_fasta)
+    tuple val(meta2), path(fasta)
+    tuple val(meta3), path(fasta_fai)
+
+    output:
+    tuple val(meta), path("*.vg") , emit: graph
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    if(vcfs && (msa && fasta)) {
+        error("Please use either VCF files and reference FASTAs or an MSA file as input.")
+    }
+
+    vcf_input = vcfs ? vcfs.collect { "--vcf ${it}" }.join(" ") : ""
+    reference = fasta ? "--reference ${fasta}" : ""
+
+    msa_input = msa ? "--msa ${msa}" : ""
+
+    insertions = insertions_fasta ? "--insertions ${insertions_fasta}" : ""
```

Comment:
Okay, I'm not familiar with this too, the docs don't match your usage and I'm having trouble accessing vg on my Mac (even docker) to view the inline help, so just to check, is this the situation?

Either:

- 'vcf mode' = vcfs, tbis, reference fasta (+fai?), plus optional insertion fa?
- 'msa mode' = msa only? with insertion fa?

So there's no 'core' set of inputs an basically everything is optional? If I've got the right end of that it feels like the msa should be a separate channel.

Reply:
That's correct!! And yeah I've thought about splitting them up but then I need to create two different inputs for `insertions_fasta`. But we can discuss this, nothing is set in stone yet :)

Reply:
strange though that it isn't working on Mac...

Reply:
Sorry, to double check- the insertions fasta applies to the msa mode as well? Despite what I might assume from:

```
 -I, --insertions FILE  a FASTA file containing insertion sequences 
                           (referred to in VCF) to add to graph.
```
?

Reply:
I haven't actually tested this, I'll do that right now!

Reply:
It does work! I've added it to the tests.